float32[] joint_data_go
float32[] joint_data_back